pagopa/io-web-profile
# [IOPID-722] Storing prev level on L3 elevation and tax code md5 in storage

## Short description

The saving of user-related information has been added when attempting to upgrade to L3 
[See the details on confluence.](https://pagopa.atlassian.net/wiki/spaces/IAEI/pages/791774211/IO-ESCO+-+Come+scalare+il+livello+di+accesso+da+L2+a+L3?src=mail&src.mail.timestamp=1694777186040&src.mail.notification=com.atlassian.confluence.plugins.confluence-content-notifications-plugin%3Apage-created-notification&src.mail.recipient=712020%3Adfce7e39-8963-4f34-8190-281ad91fad4b&atlOrigin=eyJpIjoiNjlkYzNlNDgzMGU2NGE1YmFkZGJjNmNkZTRkNzMyNWIiLCJwIjoiYyJ9&src.mail.action=view)

## How to test
On mockoon set /session-state to:

```json
{
  "access_enabled": false,
  "session_info": {
    "active": false,
    "expiration_date":"2025-10-13"
  }
}
```

Run yarn install

Run the project and attempt L2 access then click on "non hai il codice, scopri cosa fare" in order to elevate to L3 using hub spid login. 

Verify that when you return to the portal, the "privilege" object in local storage contains the previous SPID login value and identity (fiscal code MD5 hashed).

Note: The logic for using and deleting the storage is not currently present, it will be implemented in a dedicated branch/task.
